web/interceptor/helpers.go

```diff
@@ -14,45 +14,65 @@ import (
 // getAuthenticatedContext returns a new context with the user ID attached to it.
 // If the context does not contain a valid session cookie, it returns an error.
 func getAuthenticatedContext(ctx context.Context, logger *zap.SugaredLogger, tm *auth.TokenManager) (context.Context, error) {
+	claims, err := tm.GetClaims(ctx)
+	if err != nil {
+		logger.Errorf("Failed to extract claims from JWT: %v", err)
+		return nil, ErrInvalidAuthCookie
+	}
 	meta, ok := metadata.FromIncomingContext(ctx)
 	if !ok {
 		logger.Error("Failed to extract metadata")
 		return nil, ErrContextMetadata
 	}
-	token, err := extractToken(meta)
-	if err != nil {
-		logger.Errorf("Failed to extract token: %v", err)
-		return nil, err
-	}
-	claims, err := tm.GetClaims(token)
-	if err != nil {
-		logger.Errorf("Failed to extract claims from JWT: %v", err)
-		return nil, ErrInvalidAuthCookie
-	}
 	if tm.UpdateRequired(claims) {
 		logger.Debug("Updating token for user ", claims.UserID)
 		updatedToken, err := tm.NewAuthCookie(claims.UserID)
```

Comment:
Maybe this should be called `updatedCookie`? (and other references to `token` in the strings/comments if any)

Reply:
Renamed.